File: backend/src/main/java/com/aura_card/backend/exception/ErrorCode.java
package com.aura_card.backend.exception;

import lombok.Getter;
import org.springframework.http.HttpStatus;

@Getter
public enum ErrorCode {
    // System
    INTERNAL_ERROR("Lỗi hệ thống", HttpStatus.INTERNAL_SERVER_ERROR),
    UNCATEGORIZED_EXCEPTION("Lỗi chưa phân loại", HttpStatus.INTERNAL_SERVER_ERROR),

    // Auth
    EMAIL_ALREADY_EXISTS("Email đã tồn tại", HttpStatus.CONFLICT),
    INVALID_CREDENTIALS("Email hoặc mật khẩu không đúng", HttpStatus.UNAUTHORIZED),
    INVALID_PROVIDER("Tài khoản này đăng nhập bằng phương thức khác", HttpStatus.BAD_REQUEST),
    INVALID_OAUTH2_PROVIDER("Provider không hợp lệ", HttpStatus.BAD_REQUEST),
    ACCOUNT_LOCKED("Tài khoản đã bị khóa", HttpStatus.FORBIDDEN),
    UNAUTHORIZED("Chưa đăng nhập", HttpStatus.UNAUTHORIZED),
    FORBIDDEN("Không có quyền thực hiện thao tác này", HttpStatus.FORBIDDEN),

    // User
    USER_NOT_FOUND("Người dùng không tồn tại", HttpStatus.NOT_FOUND),
    GOOGLE_ACCOUNT_CANNOT_SET_PASSWORD("Tài khoản Google không thể đặt mật khẩu", HttpStatus.FORBIDDEN),
    USER_HAS_SAVED_CARD("Người dùng đã có thiệp lưu, không thể xóa", HttpStatus.BAD_REQUEST),

    // Card
    CARD_NOT_FOUND("Thiệp không tồn tại", HttpStatus.NOT_FOUND),
    CARD_NAME_ALREADY_EXISTS("Tên thiệp đã tồn tại", HttpStatus.CONFLICT),
    CARD_IMAGE_UPLOAD_FAILED("Upload hình thất bại", HttpStatus.INTERNAL_SERVER_ERROR),

    // SavedCard
    SAVED_CARD_NOT_FOUND("Thiệp lưu không tồn tại", HttpStatus.NOT_FOUND);

    private final String message;
    private final HttpStatus statusCode;

    ErrorCode(String message, HttpStatus statusCode) {
        this.message = message;
        this.statusCode = statusCode;
    }
}

File: backend-nodejs/utils/error.code.js
const ErrorCode = {
  // System
  INTERNAL_ERROR: { status: 500, message: "Lỗi hệ thống" },
  UNCATEGORIZED_EXCEPTION: { status: 500, message: "Lỗi chưa phân loại" },

  // Auth
  EMAIL_ALREADY_EXISTS: { status: 409, message: "Email đã tồn tại" },
  INVALID_CREDENTIALS: {
    status: 401,
    message: "Email hoặc mật khẩu không đúng",
  },
  INVALID_PROVIDER: { status: 400, message: "Email này đã có tài khoản" },
  INVALID_OAUTH2_PROVIDER: { status: 400, message: "Provider không hợp lệ" },
  ACCOUNT_LOCKED: { status: 403, message: "Tài khoản đã bị khóa" },
  UNAUTHORIZED: { status: 401, message: "Chưa đăng nhập" },
  FORBIDDEN: { status: 403, message: "Không có quyền thực hiện thao tác này" },

  // User
  USER_NOT_FOUND: { status: 404, message: "Người dùng không tồn tại" },
  GOOGLE_ACCOUNT_CANNOT_SET_PASSWORD: {
    status: 403,
    message: "Tài khoản Google không thể đặt mật khẩu",
  },
  USER_HAS_SAVED_CARD: {
    status: 400,
    message: "Người dùng đã có thiệp lưu, không thể xóa",
  },

  // Card
  CARD_NOT_FOUND: { status: 404, message: "Thiệp không tồn tại" },
  CARD_NAME_ALREADY_EXISTS: { status: 409, message: "Tên thiệp đã tồn tại" },
  CARD_IMAGE_UPLOAD_FAILED: { status: 500, message: "Upload hình thất bại" },
  INVALID_CARD_PAYLOAD: { status: 400, message: "Dữ liệu thiệp không hợp lệ" },

  // SavedCard
  SAVED_CARD_NOT_FOUND: { status: 404, message: "Thiệp lưu không tồn tại" },

  // File
  FRONT_IMAGE_REQUIRED: {
    status: 400,
    message: "Hình mặt trước không để trống",
  },
  INVALID_IMAGE_TYPE: {
    status: 400,
    message: "Định dạng ảnh không hợp lệ (chỉ jpg, png, webp)",
  },
  FILE_TOO_LARGE: { status: 400, message: "Kích thước ảnh tối đa 2MB" },
};

module.exports = ErrorCode;
